earthaccess/store.py

```diff
@@ -61,15 +90,53 @@ def __repr__(self) -> str:
         return repr(self.f)
 
 
+def get_cache_config(file_size: int) -> int:
+    """Determine the optimal caching configuration based on file size.
+    Note: we could even be smarter if we know the chunk sizes of the variables
+    we need to cache, e.g. using the dmrpp file and the wellknownparts cache type.
+
+    Uses 'blockcache' for all files with block sizes adjusted by file size:
+
+    - <100MB: 4MB
+    - >100MB: 4–16MB
+
+    Parameters:
+        file_size (int): Size of the file in bytes.
+
+    Returns:
+        Tuple[str, int]: A tuple containing:
+            cache_type (str): Always 'blockcache'.
+            block_size (int): Optimal block size in bytes.
+    """
+    if file_size < 100 * 1024 * 1024:
+        block_size = 4 * 1024 * 1024
+    elif 100 * 1024 * 1024 <= file_size < 1024 * 1024 * 1024:
+        block_size = 8 * 1024 * 1024
+    else:
+        block_size = 16 * 1024 * 1024
+
+    return block_size
+
+
 def _open_files(
     url_mapping: Mapping[str, Union[DataGranule, None]],
     fs: fsspec.AbstractFileSystem,
     *,
     pqdm_kwargs: Optional[Mapping[str, Any]] = None,
+    open_kwargs: Optional[Dict[str, Any]] = None,
 ) -> List[fsspec.spec.AbstractBufferedFile]:
     def multi_thread_open(data: tuple[str, Optional[DataGranule]]) -> EarthAccessFile:
         url, granule = data
-        return EarthAccessFile(fs.open(url), granule)  # type: ignore
+        f_size = fs.info(url)["size"]
+        default_cache_type = "blockcache"
+        default_block_size = get_cache_config(f_size)
+
+        open_kw = open_kwargs or {}
+        open_kw.setdefault("cache_type", default_cache_type)
+        open_kw.setdefault("block_size", default_block_size)
+
+        f = fs.open(url, **open_kw)
```

Comment:
> This open_kwargs will override the global configuration but I think this is the point

Totally agree that if `open_kwargs` is specified it should override `fsspec.config.conf`. I'm only concerned that this misses the case where `fsspec.config.conf` has relevant configuration from the user that earthaccess silently overrides when `open_kwargs` does not have `cache_type` or `block_size`. For instance, current code that does not give `open_kwargs` but does set `fsspec.config.conf` b/c until now that was the only way to configure the cache type.

```suggestion
        open_kw = open_kwargs or fsspec.config.conf or {}   # this isn't right (too general), but maybe something like this?
        open_kw.setdefault("cache_type", default_cache_type)
        open_kw.setdefault("block_size", default_block_size)

        f = fs.open(url, **open_kw)
```


You could argue we pick a better blocksize I guess ... in which case a Warning?